services/crypto_get_info.proto

```diff
@@ -59,6 +59,45 @@ message CryptoGetInfoResponse {
      */
     ResponseHeader header = 1;
 
+    message CryptoAllowance {
+        /**
+         * The account ID of the hbar allowance spender.
+         */
+        AccountID spender = 1;
+
+        /**
+         * The current balance of the spender's allowance in tinybars.
+         */
+        int64 allowance = 2;
+    }
+
+    message TokenAllowance {
+        /**
+         * The token that the allowance pertains to.
+         */
+        TokenID tokenId = 1;
+
+        /**
+         * The account ID of the token allowance spender.
+         */
+        AccountID spender = 2;
+
+        /**
+         * The current balance of the spender's token allowance.
+         */
+        int64 allowance = 3;
+
+        /**
+         * The number of decimal places the token allowance value is divisible by.
+         */
+        int32 decimals = 4;
+
+        /**
+         * If true, the spender has access to all of the account owner's NFT instances (currently
+         * owned and any in the future).
+         */
+        bool approvedForAll = 5;
```

Comment:
Bit confusing that the transaction has this as a `oneof` but it's okay to show it as flattened in the query response. We should try to be consistent. I'm wondering if the `oneof` in the transaction is truly needed as the default of `false` works and any invalid input can throw an error.

Reply:
As per discussion, left fields denormalized but made them wrapper types to accommodate case where allowance is not applicable (approvedForAll with NFT) and approvedForAll is not applicable (fungible token).